src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeOperator.scala

```diff
@@ -307,15 +318,28 @@ class TezosNodeOperator(val node: TezosRPCInterface, batchConf: BatchFetchConfig
       fetchedOperations <- node.runBatchedGetQuery(network, blockHashes, makeOperationsUrl, blockOperationsConcurrencyLevel)
       fetchedOperationsWithAccounts <- decodeOperations(fetchedOperations)
     } yield {
-      val operationalDataMap = fetchedOperationsWithAccounts.map{ case (hash, ops, accounts) => (hash, (ops, accounts))}.toMap
+      lazy val operationalDataMap = fetchedOperationsWithAccounts.map{ case (hash, ops, accounts) => (hash, (ops, accounts))}.toMap
       fetchedBlocksData.map {
         md =>
           val (ops, accs) = if (isGenesis(md)) (List.empty, List.empty) else operationalDataMap(md.hash)
-          (Block(md, ops), accs)
+          (parseMichelsonScripts(Block(md, ops)), accs)
       }
     }
   }
 
+  val UNPARSABLE_CODE_PLACEMENT = "Unparsable code"
+
+  private def toMichelsonScript[T <: MichelsonElement:Parser](json: Any) = {
+    Some(json).collect { case t: String => convert[T](t) } match {
+      case Some(Right(value)) => value
+      case Some(Left(t)) =>
+        logger.error(s"Error during converting Michelson format: $json", t)
```

Comment:
maybe it could help to print the `json` when we get a parsing failure?

Reply:
I don't understand. I'm printing it :)

Reply:
ahaha... I'm an idiot!